src/metal.cs

```diff
@@ -378,6 +439,32 @@ public partial interface MTLBlitCommandEncoder : MTLCommandEncoder {
 
 		[Abstract, Export ("copyFromBuffer:sourceOffset:toBuffer:destinationOffset:size:")]
 		void CopyFromBuffer (IMTLBuffer sourceBuffer, nuint sourceOffset, IMTLBuffer destinationBuffer, nuint destinationOffset, nuint size);
+		
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		// [Abstract] - should be abstract but would break the API.
+		[Export ("updateFence:")]
+		void Update (IMTLFence fence);
+
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		// [Abstract] - should be abstract but would break the API.
+		[Export ("waitForFence:")]
+		void Wait (IMTLFence fence);
+	}
+	
+	public interface IMTLFence {}
+
+	[iOS (10,0), TV (10,0), NoWatch, NoMac]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
```

Comment:
Always remove the `BaseType` if you are not going to use a `ModelAttribute`.

Rule of thumb: if no `Delegate` suffix then no `Model` or `BaseType` for it.